### PR TITLE
feat: add external blog post and update card layout

### PR DESCRIPTION
**Pull request by k8isdead**
[Preview](https://deploy-preview-238--ebpfio.netlify.app/blog)

### CODE REVIEW

File: src/common/blog-post/PostCard.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+          rel={url ? "noopener noreferrer" : ""}
+        >
+          Read More
+          <span className='link-triangle' />
+        </Link>
+        <div className='blog-tags-and-date'>
+          <span className='blog-date'>{formatPostDate(post)}</span>
+          {tags.length > 0 && (
+            <div className='blog-post-tags'>
+              {tags.map(
```
[the remark this answers is not in the document]
Comment (vannyle) [reply src/common/blog-post/PostCard.js:37]: Got it, it should be categories. Should be changed in the second iteration. 👍

**Review by vannyle — CHANGES_REQUESTED** [review]

File: gatsby-node.js [hunk 2]
```diff
@@ -91,58 +112,98 @@ exports.createPages = ({ actions, graphql }) => {
...
+      .map((item) => item.toLowerCase().replace(" ", "-"))
       .uniq()
       .value();
 
-    tags.forEach(tag => {
-      const filteredEdges = edges.filter(({ node }) => node.frontmatter.tags.map(tag => tag.toLowerCase().replace(' ', '-')).includes(tag))
+    tags.forEach((tag) => {
+      const filteredEdges = edges.filter(({ node }) =>
+        node.frontmatter.tags
+          .map((tag) => tag.toLowerCase().replace(" ", "-"))
```
Comment (vannyle) [inline gatsby-node.js:166]: `.toLowerCase().replace(" ", "-")` is repeated several times in this PR, probably I'd move it out to utils for easy reusing.

File: src/common/blog-post/Post.js [hunk 3]
```diff
@@ -1,111 +1,95 @@
...
+  const categories = post.frontmatter.categories || [];
+  const { hasPreview, previewHtml, mainHtml, previewDescription } =
+    parseHtml(post);
 
-  const html = full
-               ? mainHtml
-               : hasPreview
-                 ? previewHtml
-                 : previewDescription;
+  const html = full ? mainHtml : hasPreview ? previewHtml : previewDescription;
```
Comment (vannyle) [inline src/common/blog-post/Post.js:58]: Should we convert this double ternary condition to something more readable?

File: src/stylesheets/blog.scss [hunk 4]
```diff
@@ -181,19 +379,24 @@ $tablet: "screen and (max-width : 900px)";
...
+    background: #1a1a1a;
+    -webkit-background-clip: text;
+    -webkit-text-fill-color: transparent;
+    transition: color cubic-bezier(0.4, 0, 0.2, 1) 0.2s;
+
+    &:hover {
+      background: linear-gradient(87.89deg, #e69d00 0%, #e6b000 100%);
+      -webkit-background-clip: text;
+      -webkit-text-fill-color: transparent;
+    }
```
Comment (vannyle) [inline src/stylesheets/blog.scss:399]:
```suggestion
.blog-title, .featured-blog-title {
    font-size: 28px;
    line-height: 1.5;
    font-weight: 700;
    margin: 0;
    background: linear-gradient(87.89deg, #e69d00 0%, #e6b000 100%);
    -webkit-background-clip: text;
    background-clip: text;
    color:rgba(0,0,0,1);
    transition: color cubic-bezier(0.4, 0, 0.2, 1) 0.2s;

    &:hover {
      color:rgba(0,0,0,0);
    }
```
Currently, the transition doesn't work, removing `-webkit-text-fill-color: transparent` could fix it.

File: gatsby-node.js [hunk 5]
```diff
@@ -143,7 +146,7 @@ exports.createPages = ({ actions, graphql }) => {
       pageTemplate: path.resolve(`src/templates/blog.js`),
       paginatePost: "/blog", // old field. not remove
       pathPrefix: "/blog", // new field. not remove
-      pageLength: 8,
+      pageLength: 100,
```
Comment (vannyle) [inline gatsby-node.js:149]: It seems too many posts for one page, I would set a max of 20 posts for each page.

File: gatsby-node.js [hunk 6]
```diff
@@ -180,7 +183,7 @@ exports.createPages = ({ actions, graphql }) => {
         pageTemplate: path.resolve(`src/templates/blog.js`),
         paginatePost: `/blog/categories/${category}`, // old field. not remove
         pathPrefix: `/blog/categories/${category}`, // new field. not remove
-        pageLength: 8,
+        pageLength: 100,
```
Comment (vannyle) [inline gatsby-node.js:186]: Same here - max 20.

File: src/templates/blog.js [hunk 7]
```diff
@@ -19,53 +21,88 @@ export default function NewsIndex({ data, pageContext }) {
...
+          </>)}
+
+          <h2 className="posts-heading">Latest</h2>
+          {group.filter((post) => post.node.frontmatter.isFeatured !== true).map(({ node: post }) => (
+            <PostCard key={post.id} post={post} />
+          ))}
+            <div className='prev-next-links blog-post'>
+              <div className='prev-link'>
+                {!first && (
+                  <NavLink url={`/blog/${prevUrl}`} text='« Newer Posts' />
```
Comment (vannyle) [inline src/templates/blog.js:99]: As you return nothing, this component doesn't work properly. Should be changed to the code below. 
```
const NavLink = (props) => <Link to={props.url}>{props.text}</Link>;
```

**Review by vannyle — CHANGES_REQUESTED** [review]

**Review by vannyle — APPROVED** [review]

**Review by borkmann — APPROVED** [review]

**Pull request merged**